ontology_processing/graph_creation/process_myths.py

```diff
@@ -0,0 +1,116 @@
+import networkx as nx
+from collections import OrderedDict
+
+try:
+    from graph_creation.ontology_processing_utils import (
+        get_source_types,
+    )
+except ImportError:
+    from ontology_processing.graph_creation.ontology_processing_utils import (
+        get_source_types,
+    )
+
+class ProcessMyths:
+    """
+    Climate Myths are shown to the end user. Some myths are attached to a climate
+    solution and others are common myths not attached to a solution. Both need to
+    be added to the NetworkX object for use by the API.
+    """
+    def __init__(self, G, all_myths):
+        self.G = G
+        self.source_types = get_source_types()
+        self.general_myths = None
+        self.all_myths = all_myths
+    
+    def process_myths(self, subgraph_downstream_adaptations, nodes_upstream_greenhouse_effect):
+        """
+        Structures myth data in NetworkX object to be easier for API use.
+        """
+        general_myths = list()
+
+        for myth in self.all_myths:
+            node_neighbors = self.G.neighbors(myth)
+            for neighbor in node_neighbors:
+                if self.G[myth][neighbor]["type"] == "is_a_myth_about":
+
+                    impact_myths = []
+
+                    if "risk solution" in self.G.nodes[neighbor].keys():
+                        if "solution myths" not in self.G.nodes[neighbor].keys():
+                            solution_myths = []
+                        else:
+                            solution_myths = self.G.nodes[neighbor]["solution myths"]
+                        solution_myths.append(myth)
+                        nx.set_node_attributes(
+                            self.G, {neighbor: solution_myths}, "solution myths"
+                        )
+                    for s in subgraph_downstream_adaptations:
```

Comment:
Anything that says subgraph I think is code Henry changed and we should check how the code was before Henry changed it because his changes seem to just tangle our code up with the visualization processing code more, which makes refactoring more challenging or harder to understand?

Reply:
Okay I'm factoring all subgraph operations into the visualization file. Should be easier to understand now.